tests/system/test_misc.c

```diff
@@ -389,3 +389,102 @@ int cmd_exit_cb(int argc, char **argv)
    DEVSHELL_CMD_ASSERT(after_cb == before_cb + 1);
    return 0;
 }
+
+static bool
+comes_after(struct timeval t2,
+            struct timeval t1)
+{
+   // t1 must come before t2
+   return t2.tv_sec > t1.tv_sec || (t2.tv_sec == t1.tv_sec && t2.tv_usec > t1.tv_usec);
+}
+
+static void busy_wait(void)
+{
+   // Arbitrary amount of work
+   for (int i = 0; i < 10000; i++)
+      asmVolatile("nop"); // Don't let the compiler optimize this loop
+}
+
+static void busy_wait_kernel(void)
+{
+   int rc = syscall(TILCK_CMD_SYSCALL, TILCK_CMD_BUSY_WAIT, 10000);
+   if (rc)
```

Comment:
I'd recommend just using `DEVSHELL_CMD_ASSERT(rc == 0)` here with a simple comment on the top like: `/* TILCK_CMD_BUSY_WAIT must never fail */`

Reply:
Ok! I wrote that comment because usually I make a distinction between a test that failed and one that aborted.